prowler/providers/aws/services/iam/iam_check_saml_providers_sts/iam_check_saml_providers_sts.py

```diff
@@ -5,14 +5,23 @@
 class iam_check_saml_providers_sts(Check):
     def execute(self) -> Check_Report_AWS:
         findings = []
-        for provider in iam_client.saml_providers:
+        if iam_client.saml_providers:
+            for provider in iam_client.saml_providers:
+                report = Check_Report_AWS(self.metadata())
+                provider_name = provider["Arn"].split("/")[1]
+                report.resource_id = provider_name
+                report.resource_arn = provider["Arn"]
+                report.region = iam_client.region
+                report.status = "PASS"
+                report.status_extended = f"SAML Provider {provider_name} has been found"
+                findings.append(report)
+        else:
             report = Check_Report_AWS(self.metadata())
-            provider_name = provider["Arn"].split("/")[1]
-            report.resource_id = provider_name
-            report.resource_arn = provider["Arn"]
+            report.resource_id = ""
+            report.resource_arn = ""
             report.region = iam_client.region
-            report.status = "PASS"
-            report.status_extended = f"SAML Provider {provider_name} has been found"
+            report.status = "FAIL"
+            report.status_extended = "No SAML Providers found"
```

Comment:
Are we sure of this change? 

Also we cannot set empty both `resource_id` and `resource_arn`.

Reply:
I agree, if there are no resources, we don't create a finding.

Reply:
Changed